executor/src/main/java/org/k8loud/executor/util/ExecutorAspect.java

```diff
@@ -0,0 +1,80 @@
+package org.k8loud.executor.util;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.k8loud.executor.exception.CustomException;
+import org.k8loud.executor.util.annotation.ThrowExceptionAndLogExecutionTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+
+@Component
+@Aspect
+public class ExecutorAspect {
```

Comment:
what kind of black magic is this?

Reply:
Aspect class implements methods which are executed when our annotation is used